![Banner|50](Assets/Banner_ETL.png)

--- 
Librerias
---


In [1]:
import pandas as pd
import numpy as np
import pymysql
# from summarytools import dfSummary

---
Carga de datos en dataFrames
---

In [82]:
df_AcVelLoc_SR = pd.read_excel("Data/Internet.xlsx", sheet_name='Acc_vel_loc_sinrangos')
df_AcVel_SR = pd.read_excel("Data/Internet.xlsx", sheet_name='Velocidad_sin_Rangos')
df_AcTec_Loc = pd.read_excel("Data/Internet.xlsx", sheet_name='Accesos_tecnologia_localidad')
df_VelPorc_x_Prov = pd.read_excel("Data/Internet.xlsx", sheet_name='Velocidad % por prov')
df_Tot_VMD = pd.read_excel("Data/Internet.xlsx", sheet_name='Totales VMD')
df_TotAccTecn = pd.read_excel("Data/Internet.xlsx", sheet_name='Totales Accesos Por Tecnología')
df_Acc_x_Tec = pd.read_excel("Data/Internet.xlsx", sheet_name='Accesos Por Tecnología')
df_Pen_Pob = pd.read_excel("Data/Internet.xlsx", sheet_name='Penetración-poblacion')
df_Pen_Hog = pd.read_excel("Data/Internet.xlsx", sheet_name='Penetracion-hogares')
df_Pen_Tot = pd.read_excel("Data/Internet.xlsx", sheet_name='Penetracion-totales')
df_Tot_Acc_x_Vel = pd.read_excel("Data/Internet.xlsx", sheet_name='Totales Accesos por velocidad')
df_Acc_x_Vel = pd.read_excel("Data/Internet.xlsx", sheet_name='Accesos por velocidad')
df_Dial_BAf = pd.read_excel("Data/Internet.xlsx", sheet_name='Dial-BAf')
df_Tot_Dial_BAf = pd.read_excel("Data/Internet.xlsx", sheet_name='Totales Dial-BAf')
df_Ingresos = pd.read_excel("Data/Internet.xlsx", sheet_name='Ingresos ')

# Datos de Mapa de Conectividad
df_mapa_conectividad = pd.read_excel("Data/mapa_conectividad.xlsx")

---
Accesos divididos por Velocidad con Localidad Sin Rangos
---

In [83]:
df_AcVelLoc_SR.head()

Provincia     Partido          Localidad Link Indec  Otros  0,256 Mbps  \
0  BUENOS AIRES  25 de Mayo         25 de Mayo    6854100    NaN         NaN   
1  BUENOS AIRES  25 de Mayo    Agustín Mosconi    6854010    NaN         NaN   
2  BUENOS AIRES  25 de Mayo          Del Valle    6854020    NaN         NaN   
3  BUENOS AIRES  25 de Mayo          Ernestina    6854030    NaN         NaN   
4  BUENOS AIRES  25 de Mayo  Gobernador Ugarte    6854040    NaN         NaN   

   0,5 Mbps  0,512 Mbps  0,75 Mbps  1 Mbps  ...  680 Mbps  700 Mbps  800 Mbps  \
0       2.0         NaN       19.0     NaN  ...       NaN       NaN       NaN   
1       NaN         NaN        NaN     NaN  ...       NaN       NaN       NaN   
2       1.0         NaN        NaN     NaN  ...       NaN       NaN       NaN   
3       NaN         NaN        NaN     NaN  ...       NaN       NaN       NaN   
4       NaN         NaN        NaN     NaN  ...       NaN       NaN       NaN   

   850 Mbps  900 Mbps  999 Mbps  1000 Mbps  1024 Mbps  4000 Mbps  10000 Mbps  
0       NaN       NaN       NaN        NaN        NaN        NaN         NaN  
1       NaN       NaN       NaN        NaN        NaN        NaN         NaN  
2       NaN       NaN       NaN        NaN        NaN        NaN         NaN  
3       NaN       NaN       NaN        NaN        NaN        NaN         NaN  
4       NaN       NaN       NaN        NaN        NaN        NaN         NaN  

[5 rows x 125 columns]

In [84]:
df_AcVelLoc_SR.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3065 entries, 0 to 3064
Columns: 125 entries, Provincia to 10000 Mbps
dtypes: float64(121), object(4)
memory usage: 2.9+ MB


In [85]:
df_AcVelLoc_SR.describe()

Otros  0,256 Mbps      0,5 Mbps  0,512 Mbps    0,75 Mbps  \
count   1030.000000   36.000000    642.000000  525.000000   284.000000   
mean     318.614563    2.944444     42.978193    2.100952    39.588028   
std     2964.939611    5.806127    985.466627    4.684645   118.563946   
min      -15.000000    1.000000      0.000000    1.000000     1.000000   
25%        1.000000    1.000000      1.000000    1.000000     3.000000   
50%        6.000000    1.000000      1.000000    1.000000     9.500000   
75%       48.000000    2.000000      2.000000    1.000000    28.000000   
max    87722.000000   31.000000  24968.000000   61.000000  1307.000000   

            1 Mbps     1,5 Mbps       2 Mbps  2,2 Mbps    2,5 Mbps  ...  \
count   872.000000    15.000000   677.000000       1.0    4.000000  ...   
mean     66.642202   132.533333   101.428360      18.0  156.500000  ...   
std     222.345674   291.222268   278.821468       NaN  212.528429  ...   
min       0.000000     7.000000     1.000000      18.0   21.000000  ...   
25%       2.000000    31.000000     4.000000      18.0   35.250000  ...   
50%      10.000000    47.000000    21.000000      18.0   66.500000  ...   
75%      40.250000    75.000000    77.000000      18.0  187.750000  ...   
max    4111.000000  1170.000000  3530.000000      18.0  472.000000  ...   

       680 Mbps  700 Mbps    800 Mbps  850 Mbps  900 Mbps   999 Mbps  \
count       1.0       4.0    4.000000       1.0       1.0  17.000000   
mean        5.0       1.0  153.500000       1.0     110.0   3.352941   
std         NaN       0.0  273.381906       NaN       NaN   5.700232   
min         5.0       1.0    1.000000       1.0     110.0   1.000000   
25%         5.0       1.0   10.750000       1.0     110.0   1.000000   
50%         5.0       1.0   25.000000       1.0     110.0   2.000000   
75%         5.0       1.0  167.750000       1.0     110.0   3.000000   
max         5.0       1.0  563.000000       1.0     110.0  25.000000   

          1000 Mbps  1024 Mbps  4000 Mbps  10000 Mbps  
count    420.000000        2.0        1.0    2.000000  
mean     609.683333        1.0        1.0    5.000000  
std     3431.387543        0.0        NaN    5.656854  
min        1.000000        1.0        1.0    1.000000  
25%       17.000000        1.0        1.0    3.000000  
50%      139.500000        1.0        1.0    5.000000  
75%      498.000000        1.0        1.0    7.000000  
max    68405.000000        1.0        1.0    9.000000  

[8 rows x 121 columns]

In [86]:
print ('cantidad nulos por columna:')
for columna in df_AcVelLoc_SR.columns:
    print(f"-->Columna: {columna}, Total nulos: {df_AcVelLoc_SR[columna].isnull().sum()}")

cantidad nulos por columna:
-->Columna: Provincia, Total nulos: 0
-->Columna: Partido, Total nulos: 0
-->Columna: Localidad, Total nulos: 0
-->Columna: Link Indec, Total nulos: 0
-->Columna: Otros, Total nulos: 2035
-->Columna: 0,256 Mbps, Total nulos: 3029
-->Columna: 0,5 Mbps, Total nulos: 2423
-->Columna: 0,512 Mbps, Total nulos: 2540
-->Columna: 0,75 Mbps, Total nulos: 2781
-->Columna: 1 Mbps, Total nulos: 2193
-->Columna: 1,5 Mbps, Total nulos: 3050
-->Columna: 2 Mbps, Total nulos: 2388
-->Columna: 2,2 Mbps, Total nulos: 3064
-->Columna: 2,5 Mbps, Total nulos: 3061
-->Columna: 3 Mbps, Total nulos: 1801
-->Columna: 3,3 Mbps, Total nulos: 3064
-->Columna: 3,5 Mbps, Total nulos: 2695
-->Columna: 4 Mbps, Total nulos: 2592
-->Columna: 4,5 Mbps, Total nulos: 3064
-->Columna: 5 Mbps, Total nulos: 2105
-->Columna: 6 Mbps, Total nulos: 1718
-->Columna: 6,4 Mbps, Total nulos: 3064
-->Columna: 7 Mbps, Total nulos: 2872
-->Columna: 7,5 Mbps, Total nulos: 3064
-->Columna: 8 Mbps, Total nulos: 

La manera de comprimir la informacion de este Dataframe es mediante la agrupación de columnas por rangos de velocidad. De esta manera, se hace más legible la información.

In [93]:
# Se agrupa el Dataframe de la página 1 mediante rangos de Mbps
# Se definen los rangos de columnas
column_groups = {
    '0 a 1 Mbps': ['0,256 Mbps', '0,5 Mbps', '0,512 Mbps', '0,75 Mbps', '1 Mbps'],
    '1.5 a 3 Mbps': ['1,5 Mbps', '2 Mbps', '2,2 Mbps', '2,5 Mbps', '3 Mbps'],
    '3.3 a 5 Mbps': ['3,3 Mbps', '3,5 Mbps', '4 Mbps', '4,5 Mbps', '5 Mbps'],
    '6 a 10 Mbps': ['6 Mbps', '6,4 Mbps', '7 Mbps', '7,5 Mbps',	'8 Mbps', '9 Mbps',	'10 Mbps'],
    '10.1 a 25 Mbps': ['10,1 Mbps', '10,2 Mbps', '11 Mbps', '12 Mbps',	'12,3 Mbps', '13 Mbps',	'14 Mbps', '15 Mbps', '16 Mbps', '17 Mbps', '18 Mbps', '20 Mbps', '20,5 Mbps',	'21 Mbps', '22 Mbps', '23 Mbps', '24 Mbps', '25 Mbps'],
    '25.1 a 50 Mbps': ['25,1 Mbps', '25,5 Mbps', '26 Mbps', '27 Mbps',	'30 Mbps', '31 Mbps', '32 Mbps', '35 Mbps', '36 Mbps', '40 Mbps', '41 Mbps', '45 Mbps', '48 Mbps',	'49 Mbps', '50 Mbps'],
    '50.1 a 100 Mbps': ['50,1 Mbps', '50,25 Mbps', '51 Mbps', '55 Mbps', '59 Mbps', '60 Mbps', '61 Mbps', '65 Mbps', '66 Mbps', '70 Mbps', '75 Mbps', '80 Mbps', '85 Mbps',	'90 Mbps', '95 Mbps', '98 Mbps', '100 Mbps'],
    '100.1 a 300 Mbps': ['100,1 Mbps', '100,25 Mbps', '105 Mbps', '110 Mbps', '120 Mbps', '122 Mbps', '125 Mbps', '130 Mbps', '140 Mbps', '150 Mbps', '155 Mbps', '160 Mbps', '164 Mbps', '165 Mbps', '180 Mbps', '200 Mbps', '205 Mbps' , '212 Mbps', '220 Mbps', '225 Mbps', '240 Mbps', '246 Mbps', '250 Mbps', '256 Mbps', '275 Mbps', '300 Mbps'],
    '300.1 a 500 Mbps': ['320 Mbps', '325 Mbps', '330 Mbps', '350 Mbps', '400 Mbps', '450 Mbps', '480 Mbps', '500 Mbps'],
    'Mayor a 500 Mbps': ['512 Mbps', '520 Mbps', '540 Mbps', '600 Mbps', '680 Mbps', '700 Mbps', '800 Mbps', '850 Mbps', '900 Mbps', '999 Mbps', '1000 Mbps', '1024 Mbps', '4000 Mbps', '10000 Mbps'],
    'Otros': ['Otros']
}

def group_columns_by_range(df, column_groups, agg_func='sum'):
    """
    Agrupa columnas por rangos definidos y aplica una función de agregación.
    
    :param df: DataFrame original.
    :param column_groups: Diccionario con los nombres de los grupos y las columnas correspondientes.
    :param agg_func: Función de agregación ('sum', 'mean', etc.).
    :return: DataFrame con columnas agrupadas.
    """
    grouped_df = pd.DataFrame()
    
    for group_name, columns in column_groups.items():
        if agg_func == 'sum':
            grouped_df[group_name] = df[columns].sum(axis=1)
        elif agg_func == 'mean':
            grouped_df[group_name] = df[columns].mean(axis=1)    
    return grouped_df


# Agrupando con ayuda de la función definida anteriormente
df_grouped_fn = group_columns_by_range(df_AcVelLoc_SR, column_groups)
# Se agrega la columna provincia al dataFrame
df_grouped_fn['Provincia'] = df_AcVelLoc_SR['Provincia']
# Se agrupan los datos por la columna Provincia
df_grouped_fn = df_grouped_fn.groupby(by='Provincia').sum()
# Se agrega la columna total que suma e total de cada fila para saber conexiones por Provincia
df_grouped_fn['Total'] = df_grouped_fn.sum(axis=1)

#Imprime el dataframe generado
df_grouped_fn

0 a 1 Mbps  1.5 a 3 Mbps  3.3 a 5 Mbps  6 a 10 Mbps  \
Provincia                                                                  
BUENOS AIRES            48512.0       63236.0       82510.0     308664.0   
CABA                     5934.0        4220.0       16516.0      60432.0   
CATAMARCA                 247.0        1759.0         333.0       4204.0   
CHACO                     652.0        3705.0        5667.0      13828.0   
CHUBUT                   1194.0        7512.0       21995.0      46206.0   
CORDOBA                  8155.0       33155.0       23444.0     141570.0   
CORRIENTES               2757.0        6813.0        4949.0      23405.0   
ENTRE RIOS               4397.0       11636.0        8141.0      66244.0   
FORMOSA                   243.0       11731.0        5355.0      15971.0   
JUJUY                    1326.0        4596.0        8017.0      13634.0   
LA PAMPA                 1490.0        8611.0        6979.0      46772.0   
LA RIOJA                  425.0         412.0         182.0       2308.0   
MENDOZA                  3049.0        8608.0       13391.0      34794.0   
MISIONES                 3224.0       14893.0        4879.0      33479.0   
NEUQUEN                  1133.0        6929.0       19996.0      21939.0   
RIO NEGRO                1147.0        6105.0       10658.0      26378.0   
SALTA                    1040.0        7269.0       13246.0      71367.0   
SAN JUAN                 2045.0        5009.0       12069.0      17997.0   
SAN LUIS                  688.0        3263.0        1767.0      30042.0   
SANTA CRUZ                477.0        1983.0        3777.0      11581.0   
SANTA FE                 6885.0       29557.0       23562.0     176205.0   
SANTIAGO DEL ESTERO      1314.0        5918.0        1629.0      58092.0   
TIERRA DEL FUEGO          856.0           5.0        6043.0      13795.0   
TUCUMAN                   966.0        4812.0        3340.0      27993.0   

                     10.1 a 25 Mbps  25.1 a 50 Mbps  50.1 a 100 Mbps  \
Provincia                                                              
BUENOS AIRES               284608.0       1296268.0         959048.0   
CABA                        37803.0        125198.0         394021.0   
CATAMARCA                    5165.0         36072.0          17716.0   
CHACO                        8968.0         35430.0          55941.0   
CHUBUT                      36631.0         39046.0           1257.0   
CORDOBA                     70133.0        250987.0         302237.0   
CORRIENTES                   9101.0         35132.0          37081.0   
ENTRE RIOS                  16985.0         46588.0          69813.0   
FORMOSA                      4185.0          1645.0          11492.0   
JUJUY                       28556.0         45375.0           9396.0   
LA PAMPA                    16764.0         30171.0           7900.0   
LA RIOJA                    24979.0         41437.0          11474.0   
MENDOZA                     30830.0        164518.0          28347.0   
MISIONES                     9820.0         41281.0          51214.0   
NEUQUEN                     10750.0         22966.0          30688.0   
RIO NEGRO                   24809.0         33100.0          32969.0   
SALTA                        3193.0         29347.0          38917.0   
SAN JUAN                    12563.0         29086.0           3255.0   
SAN LUIS                    79674.0          7794.0           1983.0   
SANTA CRUZ                   8667.0         14444.0           4503.0   
SANTA FE                    71117.0        201734.0         216238.0   
SANTIAGO DEL ESTERO         24299.0         12520.0          10230.0   
TIERRA DEL FUEGO            27810.0          4724.0            496.0   
TUCUMAN                     23861.0         81424.0          75389.0   

                     100.1 a 300 Mbps  300.1 a 500 Mbps  Mayor a 500 Mbps  \
Provincia                                                                   
BUENOS AIRES     

La información que se transformo en la parte de arriba, muestra rangos de velocidad de internet, sin embargo sólo del primer trimestre. comparando la informacion con el Dataframe 'Accesos por Provincia' se llega a la conclusión que es la mismal.

In [96]:
df_Acc_x_Vel.head(24)

Año  Trimestre            Provincia  HASTA 512 kbps  + 512 Kbps - 1 Mbps  \
0   2024          1         Buenos Aires         26002.0              22510.0   
1   2024          1      Capital Federal           516.0               5418.0   
2   2024          1            Catamarca            71.0                176.0   
3   2024          1                Chaco           236.0                416.0   
4   2024          1               Chubut           111.0               1083.0   
5   2024          1              Córdoba            99.0               8056.0   
6   2024          1           Corrientes            67.0               2690.0   
7   2024          1           Entre Ríos           107.0               4290.0   
8   2024          1              Formosa            85.0                158.0   
9   2024          1                Jujuy            58.0               1268.0   
10  2024          1             La Pampa            71.0               1419.0   
11  2024          1             La Rioja            18.0                407.0   
12  2024          1              Mendoza           104.0               2945.0   
13  2024          1             Misiones            51.0               3173.0   
14  2024          1              Neuquén           136.0                997.0   
15  2024          1            Río Negro           112.0               1035.0   
16  2024          1                Salta           114.0                926.0   
17  2024          1             San Juan            24.0               2021.0   
18  2024          1             San Luis            60.0                628.0   
19  2024          1           Santa Cruz            50.0                427.0   
20  2024          1             Santa Fe           468.0               6417.0   
21  2024          1  Santiago Del Estero           180.0               1134.0   
22  2024          1     Tierra Del Fuego            22.0                834.0   
23  2024          1              Tucumán            39.0                927.0   

    + 1 Mbps - 6 Mbps  + 6 Mbps - 10 Mbps  + 10 Mbps - 20 Mbps  \
0            221185.0            233225.0             241778.0   
1             29753.0             51415.0              37369.0   
2              2872.0              3424.0               4204.0   
3             14156.0              9044.0               5533.0   
4             43828.0             31885.0              23024.0   
5            110371.0             87798.0              47085.0   
6             18637.0             16530.0               7680.0   
7             34574.0             51447.0              16122.0   
8             22133.0             10924.0               4176.0   
9             14964.0             11283.0              28548.0   
10            45089.0             17273.0              15221.0   
11             1336.0              1566.0              24770.0   
12            35893.0             20900.0              30049.0   
13            26882.0             26369.0               9812.0   
14            35382.0             13482.0              10491.0   
15            25500.0             17641.0              17191.0   
16            27352.0             64530.0               3192.0   
17            22045.0             13030.0              11597.0   
18            11318.0             23754.0               8953.0   
19             7764.0              9577.0               7182.0   
20            76294.0            153030.0              58595.0   
21            12428.0             53211.0              24281.0   
22            10169.0              9674.0              12039.0   
23            16227.0             19918.0              23263.0   

    + 20 Mbps - 30 Mbps  + 30 Mbps     OTROS      Total  
0              106418.0  4068292.0  125147.0  5044557.0  
1                5957.0  1374283.0       0.0  1504711.0  
2                1822.0    62294.0      70.0    74933.0  
3                5718.0   116624.0    3816.0   155543.0  
4               20261.0    33844.0   1

Al analizar la hoja de datos del archivo que lleva por nombre -Acceso de Velocidad Sin Rangos Con Localidad- se observa lo siguiente: existen demasiados nulos en el dataframe, esto llevaría a realizar una ardua labor para conjuntar de acuerdo a localidad y rangos de Velocidad, sin embargo, existe una hoja en el archivo llamada ´Accesos por provincia´ que ya esta estructurada de esa manera por tal motivo, se procede a desechar la información de este DataFrame.

Sin embargo, las primeras cuatro columnas pueden ser de gran ayuda para formar una tabla con ´Provincia, Partido, Localidad y Link Indec´, puesto que esta información se condsidera importante para hacer consultas o en su caso unir o combinar DataFrames y nos arroja información de ubicación.


In [110]:
# Creamos un dataFrame que será el de carga a la DB
df_Acc_x_Vel_carga = df_Acc_x_Vel
# Se convierte la columna provincia a Mayusculas
df_Acc_x_Vel_carga['Provincia'] = df_Acc_x_Vel['Provincia'].str.upper()
# se analizan si hay datos Nan en el DataFrame por cada columna
df_Acc_x_Vel_carga.isna().sum()

Año                    0
Trimestre              0
Provincia              0
HASTA 512 kbps         0
+ 512 Kbps - 1 Mbps    0
+ 1 Mbps - 6 Mbps      0
+ 6 Mbps - 10 Mbps     0
+ 10 Mbps - 20 Mbps    0
+ 20 Mbps - 30 Mbps    0
+ 30 Mbps              0
OTROS                  6
Total                  0
dtype: int64

In [141]:
df_Acc_x_Vel_carga['OTROS'] = df_Acc_x_Vel_carga['OTROS'].fillna(0) # En la columna otros existen 6 nulos que se llenan con ceros ya que es conteo


Carga del DataFrame a la DB
---

In [142]:
# Que columnas tienen datos no nulos
df_Acc_x_Vel_carga.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 12 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Año                  984 non-null    int32  
 1   Trimestre            984 non-null    int32  
 2   Provincia            984 non-null    object 
 3   HASTA 512 kbps       984 non-null    int32  
 4   + 512 Kbps - 1 Mbps  984 non-null    int32  
 5   + 1 Mbps - 6 Mbps    984 non-null    int32  
 6   + 6 Mbps - 10 Mbps   984 non-null    int32  
 7   + 10 Mbps - 20 Mbps  984 non-null    int32  
 8   + 20 Mbps - 30 Mbps  984 non-null    int32  
 9   + 30 Mbps            984 non-null    int32  
 10  OTROS                984 non-null    float64
 11  Total                984 non-null    float64
dtypes: float64(2), int32(9), object(1)
memory usage: 57.8+ KB


In [143]:
df_Acc_x_Vel_carga

Año  Trimestre            Provincia  HASTA 512 kbps  \
0    2024          1         BUENOS AIRES           26002   
1    2024          1      CAPITAL FEDERAL             516   
2    2024          1            CATAMARCA              71   
3    2024          1                CHACO             236   
4    2024          1               CHUBUT             111   
..    ...        ...                  ...             ...   
979  2014          1           SANTA CRUZ             161   
980  2014          1             SANTA FE            8456   
981  2014          1  SANTIAGO DEL ESTERO            1234   
982  2014          1     TIERRA DEL FUEGO              12   
983  2014          1              TUCUMÁN               6   

     + 512 Kbps - 1 Mbps  + 1 Mbps - 6 Mbps  + 6 Mbps - 10 Mbps  \
0                  22510             221185              233225   
1                   5418              29753               51415   
2                    176               2872                3424   
3                    416              14156                9044   
4                   1083              43828               31885   
..                   ...                ...                 ...   
979                 1625              24972                   1   
980               124468             345225               20328   
981                10531              22817                2422   
982                  607              30902                   6   
983                34672              83210               11779   

     + 10 Mbps - 20 Mbps  + 20 Mbps - 30 Mbps  + 30 Mbps     OTROS      Total  
0                 241778               106418    4068292  125147.0  5044557.0  
1                  37369                 5957    1374283       0.0  1504711.0  
2                   4204                 1822      62294      70.0    74933.0  
3                   5533                 5718     116624    3816.0   155543.0  
4                  23024                20261      33844   16640.0   170676.0  
..                   ...                  ...        ...       ...        ...  
979                    1                    0          0       0.0    26760.0  
980                 6845                   23        668       0.0   506013.0  
981                  109                    0          0       0.0    37113.0  
982                    0                    0          0       0.0    31527.0  
983                  362                    3          0       0.0   130032.0  

[984 rows x 12 columns]

In [144]:
columnass = ['Año', 'Trimestre', 'HASTA 512 kbps',
       '+ 512 Kbps - 1 Mbps', '+ 1 Mbps - 6 Mbps', '+ 6 Mbps - 10 Mbps',
       '+ 10 Mbps - 20 Mbps', '+ 20 Mbps - 30 Mbps', '+ 30 Mbps', 'OTROS',
       'Total']

for colum in columnass:
    df_Acc_x_Vel_carga[colum] = df_Acc_x_Vel_carga[colum].astype(int)

In [145]:
# Datos para conectar a la DB
connection = pymysql.connect(
    host='bn83gilwrcfotqoazybx-mysql.services.clever-cloud.com',
    user='uc0tzjpbwmcv8ro5',
    password='LaNI5u2ZjXzafghQRntP',
    db='bn83gilwrcfotqoazybx'
)

In [ ]:
# Crear un cursor
cursor = connection.cursor()

# Crear una tabla (ajusta las columnas según tu dataframe)
create_table_query = '''
CREATE TABLE `bn83gilwrcfotqoazybx`.`acceso_x_velocidad` (
`anio` INT NULL , 
`trimestre` INT NULL , 
`provincia` VARCHAR(25) NULL , 
`0 a 512 Kbps` INT NULL , 
`513 Kbps a 1 Mbps` INT NULL , 
`1.1 a 6 Mbps` INT NULL , 
`6.1 a 10 Mbps` INT NULL ,
`10.1 a 20 Mbps` INT NULL , 
`20.1 a 30 Mbps` INT NULL , 
`mayor a 30 Mbps` INT NULL , 
`otros` INT NULL , 
`total` INT NULL ) ENGINE = InnoDB;
'''

cursor.execute(create_table_query)

# Confirmar los cambios
connection.commit()

# Cerrar la conexión
connection.close()

In [147]:
cursor = connection.cursor()


for i, row in df_Acc_x_Vel_carga.iterrows():
    sql = '''
    INSERT INTO `bn83gilwrcfotqoazybx`.`acceso_x_velocidad`(
    `anio`, 
    `trimestre`, 
    `provincia`, 
    `0 a 512 Kbps`, 
    `513 Kbps a 1 Mbps`, 
    `1.1 a 6 Mbps`, 
    `6.1 a 10 Mbps`, 
    `10.1 a 20 Mbps`, 
    `20.1 a 30 Mbps`, 
    `mayor a 30 Mbps`, 
    `otros`, 
    `total`) VALUES 
    (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)
    '''

    cursor.execute(sql, (row['Año'], row['Trimestre'], row['Provincia'], row['HASTA 512 kbps'], row['+ 512 Kbps - 1 Mbps'], row['+ 1 Mbps - 6 Mbps'], 
                         row['+ 6 Mbps - 10 Mbps'], row['+ 10 Mbps - 20 Mbps'], row['+ 20 Mbps - 30 Mbps'], row['+ 30 Mbps'], 
                         row['OTROS'], row['Total']))

connection.commit()

# Cerrar la conexión
connection.close()

Datos subidos a la tabla acceso_x_velocidad en la Base de Datos

---

---
Accesos divididos por Velocidad Sin Rangos
---

In [30]:
df_AcVel_SR.head()

Año  Trimestre        Provincia     Otros  0,256 Mbps  0,375 Mbps  \
0  2024.0        1.0     Buenos Aires  125147.0        18.0         0.0   
1  2024.0        1.0  Capital Federal       0.0        31.0         0.0   
2  2024.0        1.0        Catamarca      70.0         0.0         0.0   
3  2024.0        1.0            Chaco    3816.0         0.0         0.0   
4  2024.0        1.0           Chubut   16640.0         8.0         0.0   

   0,625 Mbps  0,5 Mbps  0,512 Mbps  0,75 Mbps  ...  2048 Mbps  2529 Mbps  \
0         0.0   25815.0       169.0     4611.0  ...        0.0        0.0   
1         0.0     451.0        34.0     1307.0  ...        0.0        0.0   
2         0.0      34.0        37.0        0.0  ...        0.0        0.0   
3         0.0     234.0         2.0        0.0  ...        0.0        0.0   
4         0.0      69.0        34.0      836.0  ...        0.0        0.0   

   3072 Mbps  4000 Mbps  5000 Mbps  5125 Mbps  6000 Mbps  7000 Mbps  \
0        0.0        0.0        0.0        0.0        0.0        0.0   
1        0.0        0.0        0.0        0.0        0.0        0.0   
2        0.0        0.0        0.0        0.0        0.0        0.0   
3        0.0        0.0        0.0        0.0        0.0        0.0   
4        0.0        0.0        0.0        0.0        0.0        0.0   

   8000 Mbps  10000 Mbps  
0        0.0         0.0  
1        0.0         0.0  
2        0.0         0.0  
3        0.0         0.0  
4        0.0         0.0  

[5 rows x 193 columns]

In [31]:
df_AcVel_SR.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 625 entries, 0 to 624
Columns: 193 entries, Año to 10000 Mbps
dtypes: float64(191), int64(1), object(1)
memory usage: 942.5+ KB


In [32]:
df_AcVel_SR.describe()

Año   Trimestre          Otros  0,256 Mbps  0,375 Mbps  \
count   624.000000  624.000000     624.000000  624.000000  624.000000   
mean   2020.500000    2.500000    9905.139423    3.469551    3.966346   
std       1.907996    1.152846   17973.450593   10.749551   29.632806   
min    2017.000000    1.000000       0.000000    0.000000    0.000000   
25%    2019.000000    1.000000     589.500000    0.000000    0.000000   
50%    2020.500000    2.500000    4467.500000    0.000000    0.000000   
75%    2022.000000    4.000000   13300.750000    1.000000    0.000000   
max    2024.000000    4.000000  125147.000000  102.000000  225.000000   

       0,625 Mbps       0,5 Mbps  0,512 Mbps     0,75 Mbps        1 Mbps  ...  \
count  624.000000     624.000000  624.000000    624.000000    624.000000  ...   
mean     1.956731    2236.834936   40.467949   1150.025641   4706.605769  ...   
std     14.618851   15902.906210   69.387753   4676.714406   7344.002637  ...   
min      0.000000       6.000000    0.000000      0.000000      0.000000  ...   
25%      0.000000      20.750000    0.000000      0.000000    690.250000  ...   
50%      0.000000      47.000000   16.000000      4.000000   2003.000000  ...   
75%      0.000000      82.250000   60.000000    318.000000   5196.000000  ...   
max    111.000000  238902.000000  609.000000  35185.000000  49473.000000  ...   

        2048 Mbps   2529 Mbps   3072 Mbps   4000 Mbps   5000 Mbps   5125 Mbps  \
count  624.000000  624.000000  624.000000  624.000000  624.000000  624.000000   
mean     6.903846    0.032051    1.818910    0.113782    0.059295    0.006410   
std     76.733746    0.800641   20.254423    1.209195    0.719825    0.079871   
min      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
25%      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
50%      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
75%      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
max    860.000000   20.000000  227.000000   15.000000    9.000000    1.000000   

        6000 Mbps   7000 Mbps   8000 Mbps  10000 Mbps  
count  624.000000  624.000000  624.000000  624.000000  
mean     0.054487    0.158654    0.012821    0.073718  
std      0.605151    1.441916    0.159742    0.701533  
min      0.000000    0.000000    0.000000    0.000000  
25%      0.000000    0.000000    0.000000    0.000000  
50%      0.000000    0.000000    0.000000    0.000000  
75%      0.000000    0.000000    0.000000    0.000000  
max      8.000000   18.000000    2.000000    9.000000  

[8 rows x 192 columns]

In [34]:
print ('cantidad nulos por columna:')
for columna in df_AcVel_SR.columns:
    print(f"-->Columna: {columna}, Total nulos: {df_AcVel_SR[columna].isnull().sum()}")

cantidad nulos por columna:
-->Columna: Año, Total nulos: 1
-->Columna: Trimestre, Total nulos: 1
-->Columna: Provincia, Total nulos: 1
-->Columna: Otros, Total nulos: 1
-->Columna: 0,256 Mbps, Total nulos: 1
-->Columna: 0,375 Mbps, Total nulos: 1
-->Columna: 0,625 Mbps, Total nulos: 1
-->Columna: 0,5 Mbps, Total nulos: 1
-->Columna: 0,512 Mbps, Total nulos: 1
-->Columna: 0,75 Mbps, Total nulos: 1
-->Columna: 1 Mbps, Total nulos: 1
-->Columna: 1,2 Mbps, Total nulos: 1
-->Columna: 1,25 Mbps, Total nulos: 1
-->Columna: 1,5 Mbps, Total nulos: 1
-->Columna: 2 Mbps, Total nulos: 1
-->Columna: 2,2 Mbps, Total nulos: 1
-->Columna: 2,5 Mbps, Total nulos: 0
-->Columna: 6,7 Mbps, Total nulos: 1
-->Columna: 3 Mbps, Total nulos: 1
-->Columna: 3,2 Mbps, Total nulos: 1
-->Columna: 3,3 Mbps, Total nulos: 1
-->Columna: 3,5 Mbps, Total nulos: 1
-->Columna: 4 Mbps, Total nulos: 1
-->Columna: 4,5 Mbps, Total nulos: 1
-->Columna: 5 Mbps, Total nulos: 1
-->Columna: 6 Mbps, Total nulos: 1
-->Columna: 6,4 Mb

In [43]:
print ('cantidad ceros por columna:')
conteo_ceros = (df_AcVel_SR == 0).sum()
conteo_ceros

cantidad ceros por columna:


Año             0
Trimestre       0
Provincia       0
Otros          96
0,256 Mbps    433
             ... 
5125 Mbps     620
6000 Mbps     616
7000 Mbps     615
8000 Mbps     620
10000 Mbps    614
Length: 193, dtype: int64

De igual forma que el dataFrame anterior se puede notar que los datos se deben agrupar por velocidades puesto que se tienen una gran cantidad de columnas 192, dependiendo el caso que se requiera. sin embargo del mismo modo, se tiene un dataframe procesado en los datos que es el que se utilizará para subir a la DB.
El dataframe que contiene esta información esta conjunto con el dataframe subudo anteriormente.

---

---
Totales Accesos Por Velocidad
---

In [149]:
df_Tot_Acc_x_Vel.head(5)

Año  Trimestre  Hasta 512 kbps  Entre 512 Kbps y 1 Mbps  \
0  2024          1           28801                    69355   
1  2023          4           29708                    71742   
2  2023          3           30274                    80609   
3  2023          2           30827                    79405   
4  2023          1           31437                    82048   

   Entre 1 Mbps y 6 Mbps  Entre 6 Mbps y 10 Mbps  Entre 10 Mbps y 20 Mbps  \
0                 866152                  950930                   672155   
1                 900253                  978108                   697232   
2                 945608                  998725                   720084   
3                 963510                 1028250                   734125   
4                1001553                 1060321                   729194   

   Entre 20 Mbps y 30 Mbps  Más de 30 Mbps   OTROS     Total  
0                   353896         8363694  328173  11633156  
1                   350290         8224736  295592  11547661  
2                   350918         8003269  277421  11406908  
3                   364730         7775095  275955  11251897  
4                   357769         7564819  414754  11241895

In [151]:
df_Tot_Acc_x_Vel.info() # No se observan valores nulos y todos son tipo entero

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41 entries, 0 to 40
Data columns (total 11 columns):
 #   Column                   Non-Null Count  Dtype
---  ------                   --------------  -----
 0   Año                      41 non-null     int64
 1   Trimestre                41 non-null     int64
 2   Hasta 512 kbps           41 non-null     int64
 3   Entre 512 Kbps y 1 Mbps  41 non-null     int64
 4   Entre 1 Mbps y 6 Mbps    41 non-null     int64
 5   Entre 6 Mbps y 10 Mbps   41 non-null     int64
 6   Entre 10 Mbps y 20 Mbps  41 non-null     int64
 7   Entre 20 Mbps y 30 Mbps  41 non-null     int64
 8   Más de 30 Mbps           41 non-null     int64
 9   OTROS                    41 non-null     int64
 10  Total                    41 non-null     int64
dtypes: int64(11)
memory usage: 3.7 KB


In [152]:
df_Tot_Acc_x_Vel.describe()

Año  Trimestre  Hasta 512 kbps  Entre 512 Kbps y 1 Mbps  \
count    41.000000  41.000000       41.000000                41.000000   
mean   2018.634146   2.463415    47210.390244            219239.463415   
std       2.997967   1.142313    49441.161964            193062.660919   
min    2014.000000   1.000000     5675.000000             28521.000000   
25%    2016.000000   1.000000    29708.000000             71742.000000   
50%    2019.000000   2.000000    35030.000000            195706.000000   
75%    2021.000000   3.000000    41064.000000            294413.000000   
max    2024.000000   4.000000   241713.000000            687619.000000   

       Entre 1 Mbps y 6 Mbps  Entre 6 Mbps y 10 Mbps  Entre 10 Mbps y 20 Mbps  \
count           4.100000e+01            4.100000e+01             4.100000e+01   
mean            3.233344e+06            9.155820e+05             8.898369e+05   
std             1.526298e+06            2.322244e+05             4.558204e+05   
min             8.661520e+05            2.891820e+05             1.011270e+05   
25%             2.531271e+06            8.594130e+05             6.932770e+05   
50%             3.316868e+06            9.794160e+05             7.993500e+05   
75%             4.816081e+06            1.060321e+06             1.038029e+06   
max             5.153437e+06            1.245333e+06             2.068087e+06   

       Entre 20 Mbps y 30 Mbps  Más de 30 Mbps          OTROS         Total  
count             4.100000e+01    4.100000e+01      41.000000  4.100000e+01  
mean              4.815391e+05    2.776911e+06  149513.097561  8.713176e+06  
std               4.383245e+05    3.055797e+06  130707.129029  1.738855e+06  
min               3.450000e+02    1.159500e+04       0.000000  6.272846e+06  
25%               1.245640e+05    3.102200e+04       0.000000  7.144696e+06  
50%               3.647300e+05    1.138728e+06  167369.000000  8.648135e+06  
75%               5.959200e+05    4.948174e+06  238455.000000  1.008554e+07  
max               1.688748e+06    8.363694e+06  414754.000000  1.163316e+07

In [176]:
# Por si  hay una conexion abierta --> Cerrar la conexión 
connection.close()

In [182]:
# Datos para conectar a la DB
connection = pymysql.connect(
    host='bn83gilwrcfotqoazybx-mysql.services.clever-cloud.com',
    user='uc0tzjpbwmcv8ro5',
    password='LaNI5u2ZjXzafghQRntP',
    db='bn83gilwrcfotqoazybx'
)

In [165]:
# Crear un cursor
cursor = connection.cursor()

# Crear una tabla (ajusta las columnas según tu dataframe)
create_table_query = '''
CREATE TABLE `bn83gilwrcfotqoazybx`.`totales_acceso_x_vel` (
`anio` INT NULL , 
`trimestre` INT NULL , 
`0 a 512 Kbps` INT NULL , 
`513 Kbps a 1 Mbps` INT NULL , 
`1.1 a 6 Mbps` INT NULL , 
`6.1 a 10 Mbps` INT NULL ,
`10.1 a 20 Mbps` INT NULL , 
`20.1 a 30 Mbps` INT NULL , 
`mayor a 30 Mbps` INT NULL , 
`otros` INT NULL , 
`total` INT NULL ) ENGINE = InnoDB;
'''

cursor.execute(create_table_query)

# Confirmar los cambios
connection.commit()

# Cerrar la conexión
connection.close()

In [167]:
df_Tot_Acc_x_Vel.columns

Index(['Año', 'Trimestre', 'Hasta 512 kbps', 'Entre 512 Kbps y 1 Mbps',
       'Entre 1 Mbps y 6 Mbps', 'Entre 6 Mbps y 10 Mbps',
       'Entre 10 Mbps y 20 Mbps', 'Entre 20 Mbps y 30 Mbps', 'Más de 30 Mbps',
       'OTROS', 'Total'],
      dtype='object')

In [170]:
# Mandar los datos a la DB
cursor = connection.cursor()


for i, row in df_Tot_Acc_x_Vel.iterrows():
    sql = '''
    INSERT INTO `bn83gilwrcfotqoazybx`.`totales_acceso_x_vel`(
    `anio`, 
    `trimestre`, 
    `0 a 512 Kbps`, 
    `513 Kbps a 1 Mbps`, 
    `1.1 a 6 Mbps`, 
    `6.1 a 10 Mbps`, 
    `10.1 a 20 Mbps`, 
    `20.1 a 30 Mbps`, 
    `mayor a 30 Mbps`, 
    `otros`, 
    `total`) VALUES 
    (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)
    '''

    cursor.execute(sql, (row['Año'], row['Trimestre'], row['Hasta 512 kbps'], row['Entre 512 Kbps y 1 Mbps'], row['Entre 1 Mbps y 6 Mbps'], 
                         row['Entre 6 Mbps y 10 Mbps'], row['Entre 10 Mbps y 20 Mbps'], row['Entre 20 Mbps y 30 Mbps'], row['Más de 30 Mbps'], 
                         row['OTROS'], row['Total']))

connection.commit()

# Cerrar la conexión
connection.close()

---

Velocidad Media de Bajada por provincia y trimestre/año
---


In [171]:
df_VelPorc_x_Prov.head()

Año  Trimestre        Provincia  Mbps (Media de bajada)
0  2024          1     Buenos Aires                  159.13
1  2024          1  Capital Federal                  229.94
2  2024          1        Catamarca                   91.45
3  2024          1            Chaco                  107.51
4  2024          1           Chubut                   20.50

In [172]:
df_VelPorc_x_Prov.info() # De igual forma, no se observan nulos, existen 2 enteros, 1 float y 1 string

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Año                     984 non-null    int64  
 1   Trimestre               984 non-null    int64  
 2   Provincia               984 non-null    object 
 3   Mbps (Media de bajada)  984 non-null    float64
dtypes: float64(1), int64(2), object(1)
memory usage: 30.9+ KB


In [200]:
# se convierte Provincia a Mayusculas 
df_VelPorc_x_Prov['Provincia'] = df_VelPorc_x_Prov['Provincia'].str.upper()

In [201]:
df_VelPorc_x_Prov.head(5)

Año  Trimestre        Provincia  Mbps (Media de bajada)
0  2024          1     BUENOS AIRES                  159.13
1  2024          1  CAPITAL FEDERAL                  229.94
2  2024          1        CATAMARCA                   91.45
3  2024          1            CHACO                  107.51
4  2024          1           CHUBUT                   20.50

In [212]:
# Datos para conectar a la DB
connection = pymysql.connect(
    host='bn83gilwrcfotqoazybx-mysql.services.clever-cloud.com',
    user='uc0tzjpbwmcv8ro5',
    password='LaNI5u2ZjXzafghQRntP',
    db='bn83gilwrcfotqoazybx'
)

In [208]:
# Dropea una tabla si esta existe
# Crear un cursor
cursor = connection.cursor()
create_table_query = '''
DROP TABLE IF EXISTS `bn83gilwrcfotqoazybx`.`vel_med_bajada_prov_trim`
'''
 # Ejecutar la consulta
cursor.execute(create_table_query)

# Confirmar los cambios
connection.commit()

# Cerrar la conexión
connection.close()

In [211]:
# Crear un cursor
cursor = connection.cursor()

# Crear una tabla en la DB
create_table_query = '''
CREATE TABLE `bn83gilwrcfotqoazybx`.`vel_med_bajada_prov_trim` (
`anio` INT NULL , 
`trimestre` INT NULL , 
`provincia` VARCHAR(25), 
`Mbps media_bajada` FLOAT) ENGINE = InnoDB;;
'''
 # Ejecutar la consulta
cursor.execute(create_table_query)

# Confirmar los cambios
connection.commit()

# Cerrar la conexión
connection.close()

In [213]:
# Mandar los datos a la DB
cursor = connection.cursor()


for i, row in df_VelPorc_x_Prov.iterrows():
    sql = '''
    INSERT INTO `bn83gilwrcfotqoazybx`.`vel_med_bajada_prov_trim` (
    `anio` , 
    `trimestre` , 
    `provincia` , 
    `Mbps media_bajada`) VALUES 
    (%s, %s, %s, %s)
    '''

    cursor.execute(sql, (row['Año'], row['Trimestre'], row['Provincia'], row['Mbps (Media de bajada)']))

connection.commit()

# Cerrar la conexión
connection.close()

---
Totales VDM (Detección de buzón de voz)
---
---

In [214]:
df_Tot_VMD.head() # La columna Trimestre y Trimestre.1 son similares por lo que Trimestre.1 No se utilizará

Año  Trimestre  Mbps (Media de bajada)    Trimestre.1
0  2024          1                  139.15   Ene-Mar 2024
1  2023          4                  139.04   Oct-Dic 2023
2  2023          3                  129.67  Jul-Sept 2023
3  2023          2                  123.95   Abr-Jun 2023
4  2023          1                  110.50   Ene-Mar 2023

In [215]:
df_Tot_VMD.info() # No tiene nulos, vacios o NaN,

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41 entries, 0 to 40
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Año                     41 non-null     int64  
 1   Trimestre               41 non-null     int64  
 2   Mbps (Media de bajada)  41 non-null     float64
 3   Trimestre.1             41 non-null     object 
dtypes: float64(1), int64(2), object(1)
memory usage: 1.4+ KB


In [216]:
# Datos para conectar a la DB
connection = pymysql.connect(
    host='bn83gilwrcfotqoazybx-mysql.services.clever-cloud.com',
    user='uc0tzjpbwmcv8ro5',
    password='LaNI5u2ZjXzafghQRntP',
    db='bn83gilwrcfotqoazybx'
)

# Dropea una tabla si esta existe [[[Dropea la tabla si existe]]]
# Crear un cursor
cursor = connection.cursor()
create_table_query = '''
DROP TABLE IF EXISTS `bn83gilwrcfotqoazybx`.`tot_VMD`
'''
 # Ejecutar la consulta
cursor.execute(create_table_query)
# Confirmar los cambios
connection.commit()


# Crear un cursor   [[Crea la tabla]]]
cursor = connection.cursor()
# Crear una tabla en la DB
create_table_query = '''
CREATE TABLE `bn83gilwrcfotqoazybx`.`tot_VMD` (
`anio` INT NULL , 
`trimestre` INT NULL ,  
`Mbps media_bajada` FLOAT) ENGINE = InnoDB;;
'''
 # Ejecutar la consulta
cursor.execute(create_table_query)
# Confirmar los cambios
connection.commit()



# Mandar los datos a la DB [[Cargar DATA]]
cursor = connection.cursor()
for i, row in df_Tot_VMD.iterrows():
    sql = '''
    INSERT INTO `bn83gilwrcfotqoazybx`.`tot_VMD` (
    `anio` , 
    `trimestre` ,  
    `Mbps media_bajada`) VALUES 
    (%s, %s, %s)
    '''
    cursor.execute(sql, (row['Año'], row['Trimestre'], row['Mbps (Media de bajada)']))

connection.commit()

# Cerrar la conexión
connection.close()



---
Totales Accesos por Tecnología
---
---

In [217]:
df_TotAccTecn.head(5)

Año  Trimestre     ADSL  Cablemodem  Fibra óptica  Wireless   Otros  \
0  2024          1   774475     5986957       4015101    598682  257941   
1  2023          4   836390     6022532       3908183    585760  194796   
2  2023          3   897895     6018832       3708718    581436  200027   
3  2023          2  1006509     5997149       3463988    581823  202428   
4  2023          1  1204417     6002582       3243411    576647  214838   

      Total        Periodo  
0  11633156   Ene-Mar 2024  
1  11547661   Oct-Dic 2023  
2  11406908  Jul-Sept 2023  
3  11251897   Abr-Jun 2023  
4  11241895   Ene-Mar 2023

In [219]:
df_TotAccTecn.info() # No hay datos nulos, nan o vacios, por loq ue se mandan a la DB

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41 entries, 0 to 40
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Año           41 non-null     int64 
 1   Trimestre     41 non-null     int64 
 2   ADSL          41 non-null     int64 
 3   Cablemodem    41 non-null     int64 
 4   Fibra óptica  41 non-null     int64 
 5   Wireless      41 non-null     int64 
 6   Otros         41 non-null     int64 
 7   Total         41 non-null     int64 
 8   Periodo       41 non-null     object
dtypes: int64(8), object(1)
memory usage: 3.0+ KB


In [222]:
df_TotAccTecn.describe().round(2)

Año  Trimestre        ADSL  Cablemodem  Fibra óptica   Wireless  \
count    41.00      41.00       41.00       41.00         41.00      41.00   
mean   2018.63       2.46  2742681.73  4398940.20    1152857.85  297246.78   
std       3.00       1.14  1062551.70  1316335.72    1251751.30  200464.38   
min    2014.00       1.00   774475.00  2407330.00     139187.00   70749.00   
25%    2016.00       1.00  1950631.00  3124855.00     169898.00   85370.00   
50%    2019.00       2.00  3132789.00  4435439.00     637099.00  276419.00   
75%    2021.00       3.00  3714764.00  5826257.00    1566048.00  492415.00   
max    2024.00       4.00  3803024.00  6073426.00    4015101.00  598682.00   

           Otros        Total  
count      41.00        41.00  
mean   157362.39   8749088.95  
std     83211.38   1699043.31  
min     54300.00   6398398.00  
25%     70049.00   7206950.00  
50%    178485.00   8648135.00  
75%    247016.00  10085541.00  
max    265328.00  11633156.00

In [220]:
df_TotAccTecn['Año']

0     2024
1     2023
2     2023
3     2023
4     2023
5     2022
6     2022
7     2022
8     2022
9     2021
10    2021
11    2021
12    2021
13    2020
14    2020
15    2020
16    2020
17    2019
18    2019
19    2019
20    2019
21    2018
22    2018
23    2018
24    2018
25    2017
26    2017
27    2017
28    2017
29    2016
30    2016
31    2016
32    2016
33    2015
34    2015
35    2015
36    2015
37    2014
38    2014
39    2014
40    2014
Name: Año, dtype: int64

In [228]:
# Datos para conectar a la DB
connection = pymysql.connect(
    host='bn83gilwrcfotqoazybx-mysql.services.clever-cloud.com',
    user='uc0tzjpbwmcv8ro5',
    password='LaNI5u2ZjXzafghQRntP',
    db='bn83gilwrcfotqoazybx'
)

# Dropea una tabla si esta existe [[[Dropea la tabla si existe]]]
# Crear un cursor
cursor = connection.cursor()
create_table_query = '''
DROP TABLE IF EXISTS `bn83gilwrcfotqoazybx`.`tot_acc_tecnologia`
'''
 # Ejecutar la consulta
cursor.execute(create_table_query)
# Confirmar los cambios
connection.commit()


# Crear un cursor   [[Crea la tabla]]]
cursor = connection.cursor()
# Crear una tabla en la DB
create_table_query = '''
CREATE TABLE `bn83gilwrcfotqoazybx`.`tot_acc_tecnologia` (
`anio` INT NULL , 
`trimestre` INT NULL ,  
`ADSL` INT NULL ,
`cablemodem` INT NULL ,
`fibra_optica` INT NULL ,
`wireless` INT NULL ,
`otros` INT NULL ,
`total` INT) ENGINE = InnoDB;;
'''
 # Ejecutar la consulta
cursor.execute(create_table_query)
# Confirmar los cambios
connection.commit()



# Mandar los datos a la DB [[Cargar DATA]]
cursor = connection.cursor()
for i, row in df_TotAccTecn.iterrows():
    sql = '''
    INSERT INTO `bn83gilwrcfotqoazybx`.`tot_acc_tecnologia` (
    `anio` , 
    `trimestre` ,  
    `ADSL` ,
    `cablemodem` ,
    `fibra_optica` ,
    `wireless` ,
    `otros`,
    `total`) VALUES 
    (%s, %s, %s, %s, %s, %s, %s, %s)
    '''
    cursor.execute(sql, (row['Año'], row['Trimestre'], row['ADSL'], row['Cablemodem'], row['Fibra óptica'], row['Wireless'], row['Otros'], row['Total']))

connection.commit()

# Cerrar la conexión
connection.close()


---
Accesos Por Tecnología
---
---


In [229]:
df_Acc_x_Tec.head(5)

Año Trimestre        Provincia      ADSL  Cablemodem  Fibra óptica  \
0  2024         1     Buenos Aires  203297.0   2790663.0     1799013.0   
1  2024         1  Capital Federal   70108.0   1177475.0      203756.0   
2  2024         1        Catamarca    5901.0     11025.0       56542.0   
3  2024         1            Chaco   11343.0     58929.0       64463.0   
4  2024         1           Chubut   26306.0     77217.0       26356.0   

   Wireless     Otros      Total  
0  150290.0  101294.0  5044557.0  
1    4493.0   48879.0  1504711.0  
2    1384.0      81.0    74933.0  
3    8404.0   12404.0   155543.0  
4   30867.0    9930.0   170676.0

In [230]:
df_Acc_x_Tec.info() #Se observan datos nulos

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 986 entries, 0 to 985
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Año           985 non-null    object 
 1   Trimestre     985 non-null    object 
 2   Provincia     984 non-null    object 
 3   ADSL          984 non-null    float64
 4   Cablemodem    984 non-null    float64
 5   Fibra óptica  984 non-null    float64
 6   Wireless      984 non-null    float64
 7   Otros         984 non-null    float64
 8   Total         984 non-null    float64
dtypes: float64(6), object(3)
memory usage: 69.5+ KB


In [233]:
df_Acc_x_Tec['Año']

0      2024
1      2024
2      2024
3      2024
4      2024
       ... 
981    2014
982    2014
983    2014
984     NaN
985       *
Name: Año, Length: 986, dtype: object

In [242]:
df_Acc_x_Tec.loc[984] # La fila con indice 984 tiene datos NaN por lo que se procede a eliminar 

Año             NaN
Trimestre       NaN
Provincia       NaN
ADSL            NaN
Cablemodem      NaN
Fibra óptica    NaN
Wireless        NaN
Otros           NaN
Total           NaN
Name: 984, dtype: object

In [241]:
df_Acc_x_Tec.loc[985] # La fila con indice 985 tiene datos NaN por lo que se procede a eliminar 

Año                                                             *
Trimestre       Los datos provinciales no coinciden a nivel na...
Provincia                                                     NaN
ADSL                                                          NaN
Cablemodem                                                    NaN
Fibra óptica                                                  NaN
Wireless                                                      NaN
Otros                                                         NaN
Total                                                         NaN
Name: 985, dtype: object

In [243]:
df_Acc_x_Tec.drop([984,985], axis=0, inplace=True)

In [244]:
df_Acc_x_Tec

Año Trimestre            Provincia      ADSL  Cablemodem  Fibra óptica  \
0    2024         1         Buenos Aires  203297.0   2790663.0     1799013.0   
1    2024         1      Capital Federal   70108.0   1177475.0      203756.0   
2    2024         1            Catamarca    5901.0     11025.0       56542.0   
3    2024         1                Chaco   11343.0     58929.0       64463.0   
4    2024         1               Chubut   26306.0     77217.0       26356.0   
..    ...       ...                  ...       ...         ...           ...   
979  2014         1           Santa Cruz   18927.0       100.0         381.0   
980  2014         1             Santa Fe  322662.0    174296.0        3059.0   
981  2014         1  Santiago Del Estero   32567.0      3598.0          19.0   
982  2014         1     Tierra Del Fuego   21618.0      2837.0         648.0   
983  2014         1              Tucumán  129717.0        83.0         121.0   

     Wireless     Otros      Total  
0    150290.0  101294.0  5044557.0  
1      4493.0   48879.0  1504711.0  
2      1384.0      81.0    74933.0  
3      8404.0   12404.0   155543.0  
4     30867.0    9930.0   170676.0  
..        ...       ...        ...  
979    4092.0    3264.0    26764.0  
980    5951.0     644.0   506612.0  
981     915.0      34.0    37133.0  
982       1.0    2934.0    28038.0  
983      13.0      98.0   130032.0  

[984 rows x 9 columns]

In [246]:
df_Acc_x_Tec.info() #Ya no existen datos nulos

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Año           984 non-null    object 
 1   Trimestre     984 non-null    object 
 2   Provincia     984 non-null    object 
 3   ADSL          984 non-null    float64
 4   Cablemodem    984 non-null    float64
 5   Fibra óptica  984 non-null    float64
 6   Wireless      984 non-null    float64
 7   Otros         984 non-null    float64
 8   Total         984 non-null    float64
dtypes: float64(6), object(3)
memory usage: 69.3+ KB


In [249]:
df_Acc_x_Tec['Año'] = df_Acc_x_Tec['Año'].astype(int)

ValueError: invalid literal for int() with base 10: '2019 *'

In [250]:
# Existe un dato del 2019 que tiene 2019 *, por lo que se procede a cambiarlo
df_Acc_x_Tec.loc[df_Acc_x_Tec['Año'] == '2019 *'] = 2019

In [253]:
# Se convierte la columna a tipo entero
df_Acc_x_Tec['Año'] = df_Acc_x_Tec['Año'].astype(int)

In [254]:
# Se comprueba la transformación
df_Acc_x_Tec.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Año           984 non-null    int32  
 1   Trimestre     984 non-null    object 
 2   Provincia     984 non-null    object 
 3   ADSL          984 non-null    float64
 4   Cablemodem    984 non-null    float64
 5   Fibra óptica  984 non-null    float64
 6   Wireless      984 non-null    float64
 7   Otros         984 non-null    float64
 8   Total         984 non-null    float64
dtypes: float64(6), int32(1), object(2)
memory usage: 65.5+ KB


In [247]:
# Se transforman a mayusculas
df_Acc_x_Tec['Provincia'] = df_Acc_x_Tec['Provincia'].str.upper()

In [264]:
df_Acc_x_Tec['Provincia']

0             BUENOS AIRES
1          CAPITAL FEDERAL
2                CATAMARCA
3                    CHACO
4                   CHUBUT
              ...         
979             SANTA CRUZ
980               SANTA FE
981    SANTIAGO DEL ESTERO
982       TIERRA DEL FUEGO
983                TUCUMÁN
Name: Provincia, Length: 984, dtype: object

In [259]:
df_Acc_x_Tec.columns

Index(['Año', 'Trimestre', 'Provincia', 'ADSL', 'Cablemodem', 'Fibra óptica',
       'Wireless', 'Otros', 'Total'],
      dtype='object')

In [270]:
# Datos para conectar a la DB
connection = pymysql.connect(
    host='bn83gilwrcfotqoazybx-mysql.services.clever-cloud.com',
    user='uc0tzjpbwmcv8ro5',
    password='LaNI5u2ZjXzafghQRntP',
    db='bn83gilwrcfotqoazybx'
)

# Dropea una tabla si esta existe [[[Dropea la tabla si existe]]]
# Crear un cursor
cursor = connection.cursor()
create_table_query = '''
DROP TABLE IF EXISTS `bn83gilwrcfotqoazybx`.`acc_tecnologia_prov`
'''
 # Ejecutar la consulta
cursor.execute(create_table_query)
# Confirmar los cambios
connection.commit()


# Crear un cursor   [[Crea la tabla]]]
cursor = connection.cursor()
# Crear una tabla en la DB
create_table_query = '''
CREATE TABLE `bn83gilwrcfotqoazybx`.`acc_tecnologia_prov` (
`anio` INT NULL , 
`trimestre` INT NULL ,
`provincia` VARCHAR(25) ,
`ADSL` INT NULL ,
`cablemodem` INT NULL ,
`fibra_optica` INT NULL ,
`wireless` INT NULL ,
`otros` INT NULL ,
`total` INT) ENGINE = InnoDB;;
'''
 # Ejecutar la consulta
cursor.execute(create_table_query)
# Confirmar los cambios
connection.commit()



# Mandar los datos a la DB [[Cargar DATA]]
cursor = connection.cursor()
for i, row in df_Acc_x_Tec.iterrows():
    sql = '''
    INSERT INTO `bn83gilwrcfotqoazybx`.`acc_tecnologia_prov` (
    `anio` , 
    `trimestre` ,
    `provincia` ,
    `ADSL` ,
    `cablemodem` ,
    `fibra_optica` ,
    `wireless` ,
    `otros`,
    `total`) VALUES 
    (%s, %s, %s, %s, %s, %s, %s, %s, %s)
    '''
    cursor.execute(sql, (row['Año'], row['Trimestre'], row['Provincia'], row['ADSL'], row['Cablemodem'], row['Fibra óptica'], row['Wireless'], row['Otros'], row['Total']))

connection.commit()

# Cerrar la conexión
connection.close()

---
Penetración de Población por provincia y trimestre
---
---

In [271]:
df_Pen_Pob.head(5)

Año  Trimestre        Provincia  Accesos por cada 100 hab
0  2024          1     Buenos Aires                     27.80
1  2024          1  Capital Federal                     48.78
2  2024          1        Catamarca                     17.50
3  2024          1            Chaco                     12.48
4  2024          1           Chubut                     26.02

In [272]:
df_Pen_Pob.info() #No se observan datos nulos

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 4 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Año                       984 non-null    int64  
 1   Trimestre                 984 non-null    int64  
 2   Provincia                 984 non-null    object 
 3   Accesos por cada 100 hab  984 non-null    float64
dtypes: float64(1), int64(2), object(1)
memory usage: 30.9+ KB


In [273]:
# Pasar provincia a Mayusculas
df_Pen_Pob['Provincia'] = df_Pen_Pob['Provincia'].str.upper()

In [276]:
# Datos para conectar a la DB
connection = pymysql.connect(
    host='bn83gilwrcfotqoazybx-mysql.services.clever-cloud.com',
    user='uc0tzjpbwmcv8ro5',
    password='LaNI5u2ZjXzafghQRntP',
    db='bn83gilwrcfotqoazybx'
)

# Dropea una tabla si esta existe [[[Dropea la tabla si existe]]]
# Crear un cursor
cursor = connection.cursor()
create_table_query = '''
DROP TABLE IF EXISTS `bn83gilwrcfotqoazybx`.`penetracion_poblacion`
'''
 # Ejecutar la consulta
cursor.execute(create_table_query)
# Confirmar los cambios
connection.commit()


# Crear un cursor   [[Crea la tabla]]]
cursor = connection.cursor()
# Crear una tabla en la DB
create_table_query = '''
CREATE TABLE `bn83gilwrcfotqoazybx`.`penetracion_poblacion` (
`anio` INT NULL , 
`trimestre` INT NULL ,
`provincia` VARCHAR(25) ,
`accesos_cada_100_hab` FLOAT) ENGINE = InnoDB;;
'''
 # Ejecutar la consulta
cursor.execute(create_table_query)
# Confirmar los cambios
connection.commit()



# Mandar los datos a la DB [[Cargar DATA]]
cursor = connection.cursor()
for i, row in df_Pen_Pob.iterrows():
    sql = '''
    INSERT INTO `bn83gilwrcfotqoazybx`.`penetracion_poblacion` (
    `anio` , 
    `trimestre` ,
    `provincia` ,
    `accesos_cada_100_hab`) VALUES 
    (%s, %s, %s, %s)
    '''
    cursor.execute(sql, (row['Año'], row['Trimestre'], row['Provincia'], row['Accesos por cada 100 hab']))

connection.commit()

# Cerrar la conexión
connection.close()

---
Penetración Hogares por provincia y trimestre
---
---


In [277]:
df_Pen_Hog.head(5)

Año  Trimestre        Provincia  Accesos por cada 100 hogares
0  2024          1     Buenos Aires                         81.10
1  2024          1  Capital Federal                        119.53
2  2024          1        Catamarca                         68.82
3  2024          1            Chaco                         46.70
4  2024          1           Chubut                         84.89

In [278]:
df_Pen_Hog.info() # No hay datos nulos, NaN o vacios

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 4 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           984 non-null    int64  
 1   Trimestre                     984 non-null    int64  
 2   Provincia                     984 non-null    object 
 3   Accesos por cada 100 hogares  984 non-null    float64
dtypes: float64(1), int64(2), object(1)
memory usage: 30.9+ KB


In [279]:
# Pasar provincia a Mayusculas
df_Pen_Hog['Provincia'] = df_Pen_Hog['Provincia'].str.upper()

In [280]:
# Datos para conectar a la DB
connection = pymysql.connect(
    host='bn83gilwrcfotqoazybx-mysql.services.clever-cloud.com',
    user='uc0tzjpbwmcv8ro5',
    password='LaNI5u2ZjXzafghQRntP',
    db='bn83gilwrcfotqoazybx'
)

# Dropea una tabla si esta existe [[[Dropea la tabla si existe]]]
# Crear un cursor
cursor = connection.cursor()
create_table_query = '''
DROP TABLE IF EXISTS `bn83gilwrcfotqoazybx`.`penetracion_hogares`
'''
 # Ejecutar la consulta
cursor.execute(create_table_query)
# Confirmar los cambios
connection.commit()


# Crear un cursor   [[Crea la tabla]]]
cursor = connection.cursor()
# Crear una tabla en la DB
create_table_query = '''
CREATE TABLE `bn83gilwrcfotqoazybx`.`penetracion_hogares` (
`anio` INT NULL , 
`trimestre` INT NULL ,
`provincia` VARCHAR(25) ,
`accesos_cada_100_hogares` FLOAT) ENGINE = InnoDB;;
'''
 # Ejecutar la consulta
cursor.execute(create_table_query)
# Confirmar los cambios
connection.commit()



# Mandar los datos a la DB [[Cargar DATA]]
cursor = connection.cursor()
for i, row in df_Pen_Hog.iterrows():
    sql = '''
    INSERT INTO `bn83gilwrcfotqoazybx`.`penetracion_hogares` (
    `anio` , 
    `trimestre` ,
    `provincia` ,
    `accesos_cada_100_hogares`) VALUES 
    (%s, %s, %s, %s)
    '''
    cursor.execute(sql, (row['Año'], row['Trimestre'], row['Provincia'], row['Accesos por cada 100 hogares']))

connection.commit()

# Cerrar la conexión
connection.close()

---
Penetración total
---
---

In [281]:
df_Pen_Tot.head(5)

Año  Trimestre  Accesos por cada 100 hogares  Accesos por cada 100 hab  \
0  2024          1                         78.89                     24.79   
1  2023          4                         78.56                     24.66   
2  2023          3                         77.84                     24.41   
3  2023          2                         77.02                     24.14   
4  2023          1                         77.20                     24.17   

         Periodo  
0   Ene-Mar 2024  
1   Oct-Dic 2023  
2  Jul-Sept 2023  
3   Abr-Jun 2023  
4   Ene-Mar 2023

In [282]:
df_Pen_Tot.info() # No existen datos nulos, vacios o NaN

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41 entries, 0 to 40
Data columns (total 5 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           41 non-null     int64  
 1   Trimestre                     41 non-null     int64  
 2   Accesos por cada 100 hogares  41 non-null     float64
 3   Accesos por cada 100 hab      41 non-null     float64
 4   Periodo                       41 non-null     object 
dtypes: float64(2), int64(2), object(1)
memory usage: 1.7+ KB


In [285]:
# Datos para conectar a la DB
connection = pymysql.connect(
    host='bn83gilwrcfotqoazybx-mysql.services.clever-cloud.com',
    user='uc0tzjpbwmcv8ro5',
    password='LaNI5u2ZjXzafghQRntP',
    db='bn83gilwrcfotqoazybx'
)

# Dropea una tabla si esta existe [[[Dropea la tabla si existe]]]
# Crear un cursor
cursor = connection.cursor()
create_table_query = '''
DROP TABLE IF EXISTS `bn83gilwrcfotqoazybx`.`penetracion_total`
'''
 # Ejecutar la consulta
cursor.execute(create_table_query)
# Confirmar los cambios
connection.commit()


# Crear un cursor   [[Crea la tabla]]]
cursor = connection.cursor()
# Crear una tabla en la DB
create_table_query = '''
CREATE TABLE `bn83gilwrcfotqoazybx`.`penetracion_total` (
`anio` INT NULL , 
`trimestre` INT NULL ,
`acceso_cada_100_hogares` FLOAT ,
`accesos_cada_100_hab` FLOAT) ENGINE = InnoDB;;
'''
 # Ejecutar la consulta
cursor.execute(create_table_query)
# Confirmar los cambios
connection.commit()



# Mandar los datos a la DB [[Cargar DATA]]
cursor = connection.cursor()
for i, row in df_Pen_Tot.iterrows():
    sql = '''
    INSERT INTO `bn83gilwrcfotqoazybx`.`penetracion_total` (
    `anio` , 
    `trimestre` ,
    `acceso_cada_100_hogares` ,
    `accesos_cada_100_hab`) VALUES 
    (%s, %s, %s, %s)
    '''
    cursor.execute(sql, (row['Año'], row['Trimestre'], row['Accesos por cada 100 hogares'], row['Accesos por cada 100 hab']))

connection.commit()

# Cerrar la conexión
connection.close()

In [283]:
df_Pen_Tot.columns

Index(['Año', 'Trimestre', 'Accesos por cada 100 hogares',
       'Accesos por cada 100 hab', 'Periodo'],
      dtype='object')

---
Dial-BAF
---
---
Este dataframe tiene información del tipo de conexción por el cual acceden los usuarios, está dividida por Provincias y trimestres, así como años en que se registró la información.
Los datos que tiene son 3: 
1. Banda Ancha Fija (BAF): Coonexión de diferentes dispositivos a la red de internet y en ella se puede transmitir voz y datos.
2. Dial UP: Concexion a internet mediante una red conectada directamente a tu dispositivo.


In [286]:
df_Dial_BAf.head(5)

Año  Trimestre        Provincia  Banda ancha fija  Dial up    Total
0  2024          1     Buenos Aires           5038657   5900.0  5044557
1  2024          1  Capital Federal           1502661   2050.0  1504711
2  2024          1        Catamarca             74932      1.0    74933
3  2024          1            Chaco            155538      5.0   155543
4  2024          1           Chubut            169819    857.0   170676

In [292]:
df_Dial_BAf.shape

(984, 6)

In [287]:
df_Dial_BAf.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 6 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Año               984 non-null    int64  
 1   Trimestre         984 non-null    int64  
 2   Provincia         984 non-null    object 
 3   Banda ancha fija  984 non-null    int64  
 4   Dial up           982 non-null    float64
 5   Total             984 non-null    int64  
dtypes: float64(1), int64(4), object(1)
memory usage: 46.3+ KB


In [299]:
df_Dial_BAf[df_Dial_BAf['Dial up'] == df_Dial_BAf['Dial up'].isnull()] #Lista de datos en columna dial up con valor cero

Año  Trimestre Provincia  Banda ancha fija  Dial up   Total
8    2024          1   Formosa             62718      0.0   62718
9    2024          1     Jujuy            118053      0.0  118053
13   2024          1  Misiones            202556      0.0  202556
32   2023          4   Formosa             62002      0.0   62002
33   2023          4     Jujuy            118020      0.0  118020
..    ...        ...       ...               ...      ...     ...
728  2016          3   Formosa             27716      0.0   27716
733  2016          3  Misiones             97997      0.0   97997
925  2014          3  Misiones             75327      0.0   75327
949  2014          2  Misiones             74689      0.0   74689
973  2014          1  Misiones             73950      0.0   73950

[80 rows x 6 columns]

In [301]:
df_Dial_BAf[df_Dial_BAf['Dial up'].isnull() == True] # Filtro de datos nulos en la columna Diap up

Año  Trimestre Provincia  Banda ancha fija  Dial up   Total
560  2018          2   Formosa             44554      NaN   44554
565  2018          2  Misiones            106147      NaN  106147

In [304]:
df_Dial_BAf['Dial up'] = df_Dial_BAf['Dial up'].fillna(0) # Se llenan los nulos con cero 

In [308]:
df_Dial_BAf.info() # se comprueba la transformación 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 6 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Año               984 non-null    int64  
 1   Trimestre         984 non-null    int64  
 2   Provincia         984 non-null    object 
 3   Banda ancha fija  984 non-null    int64  
 4   Dial up           984 non-null    float64
 5   Total             984 non-null    int64  
dtypes: float64(1), int64(4), object(1)
memory usage: 46.3+ KB


In [307]:
df_Dial_BAf[df_Dial_BAf['Dial up'].isnull() == True] # se comprueba la transformación 

Empty DataFrame
Columns: [Año, Trimestre, Provincia, Banda ancha fija, Dial up, Total]
Index: []

In [314]:
# Pasar provincia a Mayusculas
df_Dial_BAf['Provincia'] = df_Dial_BAf['Provincia'].str.upper()

In [315]:
df_Dial_BAf.head(5)

Año  Trimestre        Provincia  Banda ancha fija  Dial up    Total
0  2024          1     BUENOS AIRES           5038657   5900.0  5044557
1  2024          1  CAPITAL FEDERAL           1502661   2050.0  1504711
2  2024          1        CATAMARCA             74932      1.0    74933
3  2024          1            CHACO            155538      5.0   155543
4  2024          1           CHUBUT            169819    857.0   170676

In [316]:
# Datos para conectar a la DB
connection = pymysql.connect(
    host='bn83gilwrcfotqoazybx-mysql.services.clever-cloud.com',
    user='uc0tzjpbwmcv8ro5',
    password='LaNI5u2ZjXzafghQRntP',
    db='bn83gilwrcfotqoazybx'
)

# Dropea una tabla si esta existe [[[Dropea la tabla si existe]]]
# Crear un cursor
cursor = connection.cursor()
create_table_query = '''
DROP TABLE IF EXISTS `bn83gilwrcfotqoazybx`.`dial_BAF`
'''
 # Ejecutar la consulta
cursor.execute(create_table_query)
# Confirmar los cambios
connection.commit()


# Crear un cursor   [[Crea la tabla]]]
cursor = connection.cursor()
# Crear una tabla en la DB
create_table_query = '''
CREATE TABLE `bn83gilwrcfotqoazybx`.`dial_BAF` (
`anio` INT NULL , 
`trimestre` INT NULL ,
`provincia` VARCHAR(30) ,
`banda_ancha_fija` INT NULL ,
`total` INT NULL) ENGINE = InnoDB;;
'''
 # Ejecutar la consulta
cursor.execute(create_table_query)
# Confirmar los cambios
connection.commit()



# Mandar los datos a la DB [[Cargar DATA]]
cursor = connection.cursor()
for i, row in df_Dial_BAf.iterrows():
    sql = '''
    INSERT INTO `bn83gilwrcfotqoazybx`.`dial_BAF` (
    `anio` , 
    `trimestre` ,
    `provincia` ,
    `banda_ancha_fija` ,
    `total`) VALUES 
    (%s, %s, %s, %s, %s)
    '''
    cursor.execute(sql, (row['Año'], row['Trimestre'], row['Provincia'], row['Banda ancha fija'], row['Dial up']))

connection.commit()

# Cerrar la conexión
connection.close()

In [309]:
df_Dial_BAf.columns

Index(['Año', 'Trimestre', 'Provincia', 'Banda ancha fija', 'Dial up',
       'Total'],
      dtype='object')

---
Ingresos
---
---
Este dataframe tiene información del tipo de los Ingresos en miles de pesos por trimestre o periodo y por año que generan las telecomunicaciones en Argentina
Los datos que tiene son 4: 
1. Año: Indica el año en que se registro el ingreso.
2. Trimestre: Indica el trimestre del año anterior que se registró el ingreso.
1. Ingresos (miles de pesos): cantidad monetaria de ingreso en miles de pesos que se genero en un periodo y año específico.
2. Periodo: Indica el periodo de ingreso.

In [318]:
df_Ingresos.head()

Año  Trimestre  Ingresos (miles de pesos)        Periodo
0  2024          1               2.804191e+08   Ene-Mar 2024
1  2023          4               1.673760e+08   Oct-Dic 2023
2  2023          3               1.331066e+08  Jul-Sept 2023
3  2023          2               1.180603e+08   Jun-Mar 2023
4  2033          1               1.041350e+08   Ene-Mar 2023

In [320]:
df_Ingresos.info() # se tienen 4 columnas 2 de tipo int, 1 de tipo float y 1 de tipo string

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41 entries, 0 to 40
Data columns (total 4 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Año                        41 non-null     int64  
 1   Trimestre                  41 non-null     int64  
 2   Ingresos (miles de pesos)  41 non-null     float64
 3   Periodo                    41 non-null     object 
dtypes: float64(1), int64(2), object(1)
memory usage: 1.4+ KB


Se podria pensar que el año deba transformarse a Datetime sin embargo esto ocuparía más espacio en la DB, se procede a realizar la transformacion en Power Query

In [331]:
# Datos para conectar a la DB
connection = pymysql.connect(
    host='bn83gilwrcfotqoazybx-mysql.services.clever-cloud.com',
    user='uc0tzjpbwmcv8ro5',
    password='LaNI5u2ZjXzafghQRntP',
    db='bn83gilwrcfotqoazybx'
)

# Dropea una tabla si esta existe [[[Dropea la tabla si existe]]]
# Crear un cursor
cursor = connection.cursor()
create_table_query = '''
DROP TABLE IF EXISTS `bn83gilwrcfotqoazybx`.`ingresos`
'''
 # Ejecutar la consulta
cursor.execute(create_table_query)
# Confirmar los cambios
connection.commit()


# Crear un cursor   [[Crea la tabla]]]
cursor = connection.cursor()
# Crear una tabla en la DB
create_table_query = '''
CREATE TABLE `bn83gilwrcfotqoazybx`.`ingresos` (
`anio` INT NULL , 
`trimestre` INT NULL ,
`ingreso (miles de pesos)` DECIMAL(15,2)) ENGINE = InnoDB;;
'''
 # Ejecutar la consulta
cursor.execute(create_table_query)
# Confirmar los cambios
connection.commit()



# Mandar los datos a la DB [[Cargar DATA]]
cursor = connection.cursor()
for i, row in df_Ingresos.iterrows():
    sql = '''
    INSERT INTO `bn83gilwrcfotqoazybx`.`ingresos` (
    `anio` , 
    `trimestre` ,
    `ingreso (miles de pesos)`) VALUES 
    (%s, %s, %s)
    '''
    cursor.execute(sql, (row['Año'], row['Trimestre'], row['Ingresos (miles de pesos)']))

connection.commit()

# Cerrar la conexión
connection.close()

In [332]:
# Datos para conectar a la DB
connection = pymysql.connect(
    host='bn83gilwrcfotqoazybx-mysql.services.clever-cloud.com',
    user='uc0tzjpbwmcv8ro5',
    password='LaNI5u2ZjXzafghQRntP',
    db='bn83gilwrcfotqoazybx'
)

# Dropea una tabla si esta existe [[[Dropea la tabla si existe]]]
# Crear un cursor
cursor = connection.cursor()
create_table_query = '''
DROP TABLE IF EXISTS `bn83gilwrcfotqoazybx`.`Ingresos`
'''
 # Ejecutar la consulta
cursor.execute(create_table_query)
# Confirmar los cambios
connection.commit()

Todos los datos están limpios y alojados en la base de datos listos para ser consumidos por terceros o para proyectos, sin embargo para poder hacer uniones entre las diferentes tablas que se alojaron, es necesario crear dos tablas más que son de mucha de importancia:
1. mapa de conectividad: la cual contiene datos de provincias, por partido y localidad así como coordenadas para gráficos de mapa.
2. periodicidad: esta tabla contiene información adicional de los trimestres separados por cada año.

---
Mapa de conectividad
---
---

In [333]:
df_mapa_conectividad.head(5)

Provincia     Partido          Localidad  Población ADSL Cablemódem  \
0  BUENOS AIRES  25 de Mayo         25 de Mayo      23408   SI         SI   
1  BUENOS AIRES  25 de Mayo    Agustín Mosconi        280   --         --   
2  BUENOS AIRES  25 de Mayo          Del Valle        899   SI         --   
3  BUENOS AIRES  25 de Mayo          Ernestina        145   SI         --   
4  BUENOS AIRES  25 de Mayo  Gobernador Ugarte        547   --         --   

  Dial Up Fibra óptica Satelital Wireless Telefonía Fija  3G  4G       Link  \
0      --           --        SI       SI             SI  SI  SI  6854100.0   
1      --           --        --       --             SI  --  --  6854010.0   
2      --           --        SI       --             --  SI  --  6854020.0   
3      --           --        --       --             SI  --  --  6854030.0   
4      --           --        --       SI             --  --  --  6854040.0   

     Latitud   Longitud  
0 -35.433939 -60.173121  
1 -35.739217 -60.559063  
2 -35.897345 -60.731609  
3 -35.270234 -59.557503  
4 -35.164397 -60.081358

In [334]:
df_mapa_conectividad.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4305 entries, 0 to 4304
Data columns (total 16 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Provincia       4305 non-null   object 
 1   Partido         4305 non-null   object 
 2   Localidad       4305 non-null   object 
 3   Población       4305 non-null   int64  
 4   ADSL            4305 non-null   object 
 5   Cablemódem      4305 non-null   object 
 6   Dial Up         4305 non-null   object 
 7   Fibra óptica    4305 non-null   object 
 8   Satelital       4305 non-null   object 
 9   Wireless        4305 non-null   object 
 10  Telefonía Fija  4305 non-null   object 
 11  3G              4305 non-null   object 
 12  4G              4305 non-null   object 
 13  Link            4293 non-null   float64
 14  Latitud         4305 non-null   float64
 15  Longitud        4305 non-null   float64
dtypes: float64(3), int64(1), object(12)
memory usage: 538.3+ KB


In [348]:
# Pasar provincia a Mayusculas
df_mapa_conectividad['Partido'] = df_mapa_conectividad['Partido'].str.upper()
df_mapa_conectividad['Localidad'] = df_mapa_conectividad['Localidad'].str.upper()

In [349]:
df_mapa_conectividad

Provincia           Partido                Localidad  Población  \
0     BUENOS AIRES        25 DE MAYO               25 DE MAYO      23408   
1     BUENOS AIRES        25 DE MAYO          AGUSTÍN MOSCONI        280   
2     BUENOS AIRES        25 DE MAYO                DEL VALLE        899   
3     BUENOS AIRES        25 DE MAYO                ERNESTINA        145   
4     BUENOS AIRES        25 DE MAYO        GOBERNADOR UGARTE        547   
...            ...               ...                      ...        ...   
4300      SANTA FE     GENERAL LÓPEZ              WHEELWRIGHT       5733   
4301      SANTA FE  GENERAL OBLIGADO            ARROYO CEIBAL        622   
4302      SANTA FE  GENERAL OBLIGADO  AVELLANEDA (EST. EWALD)      23341   
4303      SANTA FE  GENERAL OBLIGADO                    BERNA        629   
4304      SANTA FE  GENERAL OBLIGADO                 EL ARAZA        550   

      ADSL  Cablemódem  Dial Up  Fibra óptica  Satelital  Wireless  \
0        1           1        0             0          1         1   
1        0           0        0             0          0         0   
2        1           0        0             0          1         0   
3        1           0        0             0          0         0   
4        0           0        0             0          0         1   
...    ...         ...      ...           ...        ...       ...   
4300     0           1        0             1          1         1   
4301     0           0        0             0          0         1   
4302     1           0        0             1          0         1   
4303     0           0        0             0          0         1   
4304     0           0        0             0          0         1   

      Telefonía Fija  3G  4G        Link    Latitud   Longitud  
0                  1   1   1   6854100.0 -35.433939 -60.173121  
1                  1   0   0   6854010.0 -35.739217 -60.559063  
2                  0   1   0   6854020.0 -35.897345 -60.731609  
3                  1   0   0   6854030.0 -35.270234 -59.557503  
4                  0   0   0   6854040.0 -35.164397 -60.081358  
...              ...  ..  ..         ...        ...        ...  
4300               1   1   1  82042310.0 -33.794294 -61.211441  
4301               1   0   1  82049010.0 -28.725087 -59.480416  
4302               1   1   1  82049020.0 -29.119366 -59.659251  
4303               1   0   1  82049030.0 -29.275268 -59.847222  
4304               1   0   1  82049040.0 -29.133494 -59.947315  

[4305 rows x 16 columns]

In [350]:
df_mapa_conectividad = df_mapa_conectividad.replace({'SI': 1, '--': 0})

In [351]:
df_mapa_conectividad.head(5)

Provincia     Partido          Localidad  Población  ADSL  Cablemódem  \
0  BUENOS AIRES  25 DE MAYO         25 DE MAYO      23408     1           1   
1  BUENOS AIRES  25 DE MAYO    AGUSTÍN MOSCONI        280     0           0   
2  BUENOS AIRES  25 DE MAYO          DEL VALLE        899     1           0   
3  BUENOS AIRES  25 DE MAYO          ERNESTINA        145     1           0   
4  BUENOS AIRES  25 DE MAYO  GOBERNADOR UGARTE        547     0           0   

   Dial Up  Fibra óptica  Satelital  Wireless  Telefonía Fija  3G  4G  \
0        0             0          1         1               1   1   1   
1        0             0          0         0               1   0   0   
2        0             0          1         0               0   1   0   
3        0             0          0         0               1   0   0   
4        0             0          0         1               0   0   0   

        Link    Latitud   Longitud  
0  6854100.0 -35.433939 -60.173121  
1  6854010.0 -35.739217 -60.559063  
2  6854020.0 -35.897345 -60.731609  
3  6854030.0 -35.270234 -59.557503  
4  6854040.0 -35.164397 -60.081358

In [344]:
df_mapa_conectividad.info() # Aun existen datos nulos en Link, sin embargo son codigos de localidad que se llenaran con cero

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4305 entries, 0 to 4304
Data columns (total 16 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Provincia       4305 non-null   object 
 1   Partido         4305 non-null   object 
 2   Localidad       4305 non-null   object 
 3   Población       4305 non-null   int64  
 4   ADSL            4305 non-null   int64  
 5   Cablemódem      4305 non-null   int64  
 6   Dial Up         4305 non-null   int64  
 7   Fibra óptica    4305 non-null   int64  
 8   Satelital       4305 non-null   int64  
 9   Wireless        4305 non-null   int64  
 10  Telefonía Fija  4305 non-null   int64  
 11  3G              4305 non-null   int64  
 12  4G              4305 non-null   int64  
 13  Link            4293 non-null   float64
 14  Latitud         4305 non-null   float64
 15  Longitud        4305 non-null   float64
dtypes: float64(3), int64(10), object(3)
memory usage: 538.3+ KB


In [362]:
df_mapa_conectividad['Link'] = df_mapa_conectividad['Link'].fillna(0)

In [363]:
df_mapa_conectividad.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4305 entries, 0 to 4304
Data columns (total 16 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Provincia       4305 non-null   object 
 1   Partido         4305 non-null   object 
 2   Localidad       4305 non-null   object 
 3   Población       4305 non-null   int64  
 4   ADSL            4305 non-null   int64  
 5   Cablemódem      4305 non-null   int64  
 6   Dial Up         4305 non-null   int64  
 7   Fibra óptica    4305 non-null   int64  
 8   Satelital       4305 non-null   int64  
 9   Wireless        4305 non-null   int64  
 10  Telefonía Fija  4305 non-null   int64  
 11  3G              4305 non-null   int64  
 12  4G              4305 non-null   int64  
 13  Link            4305 non-null   float64
 14  Latitud         4305 non-null   float64
 15  Longitud        4305 non-null   float64
dtypes: float64(3), int64(10), object(3)
memory usage: 538.3+ KB


In [364]:
# Datos para conectar a la DB
connection = pymysql.connect(
    host='bn83gilwrcfotqoazybx-mysql.services.clever-cloud.com',
    user='uc0tzjpbwmcv8ro5',
    password='LaNI5u2ZjXzafghQRntP',
    db='bn83gilwrcfotqoazybx'
)

# Dropea una tabla si esta existe [[[Dropea la tabla si existe]]]
# Crear un cursor
cursor = connection.cursor()
create_table_query = '''
DROP TABLE IF EXISTS `bn83gilwrcfotqoazybx`.`mapa_conectividad`
'''
 # Ejecutar la consulta
cursor.execute(create_table_query)
# Confirmar los cambios
connection.commit()


# Crear un cursor   [[Crea la tabla]]]
cursor = connection.cursor()
# Crear una tabla en la DB
create_table_query = '''
CREATE TABLE `bn83gilwrcfotqoazybx`.`mapa_conectividad` (
`provincia` VARCHAR(30) , 
`partido` VARCHAR(30) ,
`localidad` VARCHAR(80) , 
`poblacion` INT NULL ,
`ADSL` VARCHAR(2) , 
`cablemodem` VARCHAR(2) , 
`dial-up` VARCHAR(2) , 
`fibra_optica` VARCHAR(2) , 
`satelital` VARCHAR(2) , 
`wireless` VARCHAR(2) ,
`tel_fija` VARCHAR(2) , 
`3G` VARCHAR(2) ,
`4G` VARCHAR(2) ,  
`link_indec` INT NULL , 
`latitud` DECIMAL(16,13) ,
`longitud` DECIMAL(16,13)) ENGINE = InnoDB;;
'''
 # Ejecutar la consulta
cursor.execute(create_table_query)
# Confirmar los cambios
connection.commit()



# Mandar los datos a la DB [[Cargar DATA]]
cursor = connection.cursor()
for i, row in df_mapa_conectividad.iterrows():
    sql = '''
    INSERT INTO `bn83gilwrcfotqoazybx`.`mapa_conectividad` (
    `provincia`, 
    `partido`  ,
    `localidad`  , 
    `poblacion` ,
    `ADSL` , 
    `cablemodem` , 
    `dial-up` , 
    `fibra_optica` , 
    `satelital` , 
    `wireless` ,
    `tel_fija` , 
    `3G` ,
    `4G` ,
    `link_indec` , 
    `latitud` ,
    `longitud`) VALUES 
    (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)
    '''
    cursor.execute(sql, (row['Provincia'], row['Partido'], row['Localidad'], row['Población'], row['ADSL'], row['Cablemódem'], row['Dial Up'], row['Fibra óptica'], 
                         row['Satelital'], row['Wireless'], row['Telefonía Fija'], row['3G'], row['4G'], row['Link'], row['Latitud'], row['Longitud']))

connection.commit()

# Cerrar la conexión
connection.close()

In [352]:
df_mapa_conectividad.columns

Index(['Provincia', 'Partido', 'Localidad', 'Población', 'ADSL', 'Cablemódem',
       'Dial Up', 'Fibra óptica', 'Satelital', 'Wireless', 'Telefonía Fija',
       '3G', '4G', 'Link', 'Latitud', 'Longitud'],
      dtype='object')

---
Ingresos por Telefonía móvil
---
---
Esta tabla contiene informacion de 4 columnas:
1. Año: Indica el año en que se tomo el dato
2. Trimestre: Contiene un numero entero que indica el numero de trimestre al que pertenece el dato
3. Numeros de SMS salientes: Catidad numérica que indica los mensajes salientes de dispositivos móviles
4. Periodo: Indica el periodo del conteo de mensajes

In [374]:
df_ing_tel_movil = pd.read_excel('Data/Telefonia_movil.xlsx', sheet_name='Ingresos')
df_acce_tel_movil = pd.read_excel('Data/Telefonia_movil.xlsx', sheet_name='Accesos')

In [368]:
df_ing_tel_movil.head()

Año  Trimestre  Ingresos (miles de $)        Periodo
0  2013          1               13135235   Ene-Mar 2013
1  2013          2               13970806   Abr-Jun 2013
2  2013          3               15235659  Jul-Sept 2013
3  2013          4               16218846   Oct-Dic 2013
4  2014          1               15938353   Ene-Mar 2013

In [369]:
df_ing_tel_movil.info() # No existen datos nulos

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45 entries, 0 to 44
Data columns (total 4 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   Año                    45 non-null     int64 
 1   Trimestre              45 non-null     int64 
 2   Ingresos (miles de $)  45 non-null     int64 
 3   Periodo                45 non-null     object
dtypes: int64(3), object(1)
memory usage: 1.5+ KB


In [372]:
# Datos para conectar a la DB
connection = pymysql.connect(
    host='bn83gilwrcfotqoazybx-mysql.services.clever-cloud.com',
    user='uc0tzjpbwmcv8ro5',
    password='LaNI5u2ZjXzafghQRntP',
    db='bn83gilwrcfotqoazybx'
)

# Dropea una tabla si esta existe [[[Dropea la tabla si existe]]]
# Crear un cursor
cursor = connection.cursor()
create_table_query = '''
DROP TABLE IF EXISTS `bn83gilwrcfotqoazybx`.`tel_movil_ingresos`
'''
 # Ejecutar la consulta
cursor.execute(create_table_query)
# Confirmar los cambios
connection.commit()


# Crear un cursor   [[Crea la tabla]]]
cursor = connection.cursor()
# Crear una tabla en la DB
create_table_query = '''
CREATE TABLE `bn83gilwrcfotqoazybx`.`tel_movil_ingresos` (
`anio` INT NULL , 
`trimestre` INT NULL ,
`ingresos(miles de pesos)` INT NULL , 
`periodo` VARCHAR(25)) ENGINE = InnoDB;;
'''
 # Ejecutar la consulta
cursor.execute(create_table_query)
# Confirmar los cambios
connection.commit()



# Mandar los datos a la DB [[Cargar DATA]]
cursor = connection.cursor()
for i, row in df_ing_tel_movil.iterrows():
    sql = '''
    INSERT INTO `bn83gilwrcfotqoazybx`.`tel_movil_ingresos` (
    `anio`, 
    `trimestre`  ,
    `ingresos(miles de pesos)`  , 
    `periodo`) VALUES 
    (%s, %s, %s, %s)
    '''
    cursor.execute(sql, (row['Año'], row['Trimestre'], row['Ingresos (miles de $)'], row['Periodo']))

connection.commit()

# Cerrar la conexión
connection.close()

---
Accesos por medio de Telefonía móvil
---
---
Esta tabla contiene informacion de 4 columnas:
1. Año: Indica el año en que se tomo el dato
2. Trimestre: Contiene un numero entero que indica el numero de trimestre al que pertenece el dato
3. Accesos pospago: Catidad numérica que indica los mensajes salientes de dispositivos móviles
4. Accesos prepago: Cantidad numérica que indica los dispositivos móviles que accesan en modo prepago 
5. Total accesos operativos: Indica la cantidad de accesos de dispositivos moviles
6. Periodo: indica el lapso de tiempo en el que fue registrado la cantidad de accesos moviles

In [375]:
df_acce_tel_movil.head()

Año  Trimestre  Total de accesos pospago  Total de accesos prepago  \
0  2013          1                 7761349.0                57147608.0   
1  2013          2                 7769110.0                58004822.0   
2  2013          3                 8002183.0                59048909.0   
3  2013          4                 8674516.0                58894516.0   
4  2014          1                 8224482.0                54725089.0   

   Total de accesos operativos        Periodo  
0                     64908957   Ene-Mar 2013  
1                     65773932   Abr-Jun 2013  
2                     67051092  Jul-Sept 2013  
3                     67569032   Oct-Dic 2013  
4                     62949571   Ene-Mar 2013

In [376]:
df_acce_tel_movil.info() #No hay datos nulos

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45 entries, 0 to 44
Data columns (total 6 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   Año                          45 non-null     int64  
 1   Trimestre                    45 non-null     int64  
 2   Total de accesos pospago     45 non-null     float64
 3   Total de accesos prepago     45 non-null     float64
 4   Total de accesos operativos  45 non-null     int64  
 5   Periodo                      45 non-null     object 
dtypes: float64(2), int64(3), object(1)
memory usage: 2.2+ KB


'Año': 'Anio',
        'Trimestre': 'Trimestre',
        'Provincia': 'Provincia',
        'HASTA 512 kbps': '0 a 512 Kbps',
        '+ 512 Kbps - 1 Mbps' : '513 Kbps a 1 Mbps',
        '+ 1 Mbps - 6 Mbps': '1.1 a 6 Mbps',
        '+ 6 Mbps - 10 Mbps': '6.1 a 10 Mbps',
        '+ 10 Mbps - 20 Mbps': '10.1 a 20 Mbps',
        '+ 20 Mbps - 30 Mbps' : '20.1 a 30 Mbps', 
        '+ 30 Mbps' : 'Mayor a 30 Mbps', 
        'OTROS': 'Otras Velocidades',
        'Total' : 'Total Velocidades', 
        'Mbps (Media de bajada)': 'Mbps (Media de bajada)',
        'ADSL' : 'ADSL', 'Cablemodem': 'Cablemodem', 'Fibra óptica':'Fibra optica',
        'Wireless': 'Wireless', 'Otros': 'Otros accesos tecnologicos', 
        'Total': 'Total accesos tecnologicos', 
        'Accesos por cada 100 hab': 'Accesos cada 100 hab',
        'Accesos por cada 100 hogares': 'Accesos cada 100 hogares', 
        'Banda ancha fija': 'Banda Ancha Fija', 'Dial up': 'Dial up', 'Total': 'Total

In [378]:
# Datos para conectar a la DB
connection = pymysql.connect(
    host='bn83gilwrcfotqoazybx-mysql.services.clever-cloud.com',
    user='uc0tzjpbwmcv8ro5',
    password='LaNI5u2ZjXzafghQRntP',
    db='bn83gilwrcfotqoazybx'
)

# Dropea una tabla si esta existe [[[Dropea la tabla si existe]]]
# Crear un cursor
cursor = connection.cursor()
create_table_query = '''
DROP TABLE IF EXISTS `bn83gilwrcfotqoazybx`.`tel_movil_accesos`
'''
 # Ejecutar la consulta
cursor.execute(create_table_query)
# Confirmar los cambios
connection.commit()


# Crear un cursor   [[Crea la tabla]]]
cursor = connection.cursor()
# Crear una tabla en la DB
create_table_query = '''
CREATE TABLE `bn83gilwrcfotqoazybx`.`tel_movil_accesos` (
`anio` INT NULL , 
`trimestre` INT NULL ,
`accesos_pospago` INT NULL , 
`accesos_prepago` INT NULL ,
`tot_acc_operativos` INT NULL ,
`periodo` VARCHAR(25)) ENGINE = InnoDB;;
'''
 # Ejecutar la consulta
cursor.execute(create_table_query)
# Confirmar los cambios
connection.commit()



# Mandar los datos a la DB [[Cargar DATA]]
cursor = connection.cursor()
for i, row in df_acce_tel_movil.iterrows():
    sql = '''
    INSERT INTO `bn83gilwrcfotqoazybx`.`tel_movil_accesos` (
    `anio`, 
    `trimestre` ,
    `accesos_pospago` , 
    `accesos_prepago` ,
    `tot_acc_operativos` , 
    `periodo`) VALUES 
    (%s, %s, %s, %s, %s, %s)
    '''
    cursor.execute(sql, (row['Año'], row['Trimestre'], row['Total de accesos pospago'], row['Total de accesos prepago'], row['Total de accesos operativos'], row['Periodo']))

connection.commit()

# Cerrar la conexión
connection.close()